core/src/com/unciv/ui/screens/worldscreen/unit/actions/UnitActions.kt

```diff
@@ -158,11 +158,41 @@ object UnitActions {
                 GUI.getMap().setCenterPosition(unit.getMovementDestination().position, true)
             })
         }
-
+        addEscortAction(unit)
         addSwapAction(unit)
         addDisbandAction(unit)
     }
 
+    private suspend fun SequenceScope<UnitAction>.addEscortAction(unit: MapUnit) {
+        // Air units cannot escort
+        if (unit.baseUnit.movesLikeAirUnits()) return
+
+        val worldScreen = GUI.getWorldScreen()
+        val selectedUnits = worldScreen.bottomUnitTable.selectedUnits
+        if (selectedUnits.size == 2) {
+            val tile = selectedUnits.first().getTile()
+            if (selectedUnits.last().getTile() != tile) return
```

Comment:
This really confuses me.

So there can be 3 different units? The one passed in and 2 selected?
Or is the unit always one of the selected?

Reply:
One of the two selected units is the original unit passed in. When we have two units we need to confirm that they are both on the same tile so that we aren't telling the original unit to start a formation with some other random unit. I will add a comment.